Automattic/wp-calypso
# Display ToS update notice

#### Changes proposed in this Pull Request

* Adds a selector that checks if user has accepted the current ToS.
* Adds an action for accepting current ToS.
* Displays a ToS update notice if user has not accepted the current ToS.
* Dispatches an action for accepting current ToS when user clicks on "Accept".

I used the same style guidelines we used on the GDPR banner (see p9mQOq-Hz-p2#comment-2068): 

Desktop | Medium | Mobile
--- | --- | ---
![image](https://user-images.githubusercontent.com/1233880/73939414-b0434580-48e9-11ea-8ac8-f01a0ebb2db6.png) | ![image](https://user-images.githubusercontent.com/1233880/73939424-b76a5380-48e9-11ea-8133-9a1c93f1b7bc.png) | ![image](https://user-images.githubusercontent.com/1233880/73939430-bc2f0780-48e9-11ea-8c24-c2ff91a3e7c9.png)

#### Testing instructions

* Comment out [this line](https://github.com/Automattic/wp-calypso/blob/a6c9835d0049ce05fe6793990cbbe159f424aa99/client/server/pages/index.js#L348) before starting Calypso locally, so the environment badge doesn't show up while testing.
* Make sure you don't have ToS marked as accepted. See D38262-code for instructions on how to reset ToS acceptance.
* Load any page in Calypso.
* Make sure the banner shows up and design looks ok in different viewports.
* Click on Accept.
* Make sure the banner is hidden and the ToS are accepted.
* Reload the current Calypso page.
* Make sure the banner doesn't show up anymore in any page.

Fixes https://github.com/Automattic/wp-calypso/issues/39046
